skops/card/_card_alternative.py

```diff
@@ -0,0 +1,361 @@
+from __future__ import annotations
+
+import re
+from dataclasses import dataclass, field
+from pathlib import Path
+from reprlib import Repr
+from typing import Any, Iterator, Protocol
+
+from huggingface_hub import CardData
+from sklearn.utils import estimator_html_repr
+from tabulate import tabulate  # type: ignore
+
+from skops.card._model_card import PlotSection, TableSection
+
+aRepr = Repr()
+aRepr.maxother = 79
+aRepr.maxstring = 79
+
+
+def split_subsection_names(key: str) -> list[str]:
+    return key.split("/")
+
+
+def _clean_table(table: str) -> str:
+    # replace line breaks "\n" with html tag <br />, however, leave end-of-line
+    # line breaks (eol_lb) intact
+    eol_lb = "|\n"
+    placeholder = "$%!?"  # arbitrary sting that never appears naturally
+    table = (
+        table.replace(eol_lb, placeholder)
+        .replace("\n", "<br />")
+        .replace(placeholder, eol_lb)
+    )
+    return table
+
+
+@dataclass
+class Section:
+    title: str
+    content: Formattable | str | None = None
+    subsections: dict[str, Section] = field(default_factory=dict)
+
+
+class Formattable(Protocol):
+    def format(self) -> str:
+        ...
+
+
+class Card:
+    @classmethod
+    def make_default(
+        cls, model, model_diagram: bool = True, metadata: CardData | None = None
+    ):
+        """Add a bunch of default sections, yet to be implemented"""
+        raise NotImplementedError
+
+    def __init__(
+        self, model, model_diagram: bool = True, metadata: CardData | None = None
+    ):
+        self.model = model
+        self.model_diagram = model_diagram
+        self.metadata = metadata or CardData()
+
+        self._data: dict[str, Section] = {}
+        self._metrics: dict[str, float | int] = {}
+        self._reset()
+
+    def _reset(self) -> None:
```

Comment:
from the code I don't see a _reset_ happening here. As in, nothing seems to be cleared.

Reply:
Ah yes, it's not used yet. The idea is that if I override, say, the model, we can internally call `reset` and more sure that everything that relates to the change is updated.

Reply:
is this going to simply erase README.md as it's still what we use to version models and create a new one? just to make sure we're on same page.

Reply:
No, this will not erase the `README.md`. The intent was just that if a user re-assigns the `model`, in the setter, we can call `self._reset()` and every part of the model card that needs updating will be updated. I will do the proper implementation once #205 is merged, I hope it becomes more obvious then.

Reply:
I feel like it makes sense to update the parts that weren't previously written by the user (hyperparameters, plot and so on) it was a pain point with `modelcards` as well.